### ISSUE TITLE
Validation loss and tensorboard ?

### ISSUE THREAD

**Issue by Shiro-LK** [fizyr/keras-retinanet#493]
Hello,

first of all, thank for the shared code, it is very useful.
I was wondering if it was possible to get the validation loss and not only the mAP validation during training?
Is it possible to see the boxes on some images with tensorboard with your implementation?

**Comment by hgaiser** [comment]
The validation loss is currently not computed in the training script, since the mAP is more telling for the state of the network. If you want to see it in tensorboard you'll have to make your own training script that computed and pushes the validation loss to tensorboard. It is currently not possible to see image outputs in tensorboard. If you find a nice and clean way to do this, it would be an interesting addition.

**Comment by Shiro-LK** [comment]
To add the validation loss, we just need to add to the fit_generator : validation_data=validation_generator and the number of steps.
Regarding the images on tensorboard, it is not very difficult, I did it. How can I add it for your repository ?

**Comment by jashshopin** [comment]
@Shiro-LK can you share your scripts?

**Comment by Shiro-LK** [comment]
I forked this repository but I also added some other parameters. Take a look at train.py (I add a parameters to Evaluate function) and the eval.py in the folder callback/utils .

**Comment by jashshopin** [comment]
@Shiro-LK I'm going through your forked repository.  What is the argument to this line? `csv_parser.add_argument('--dataset_dir', help='path to the dataset', default=None)`

**Comment by Shiro-LK** [comment]
@jashshopin It is a parameter for CSV generator. If in your CSV file you have only the name of the images (example : 001.jpg) you can state the folder where the images are (if they are all in the same folder). It is an optional parameter

**Comment by jashshopin** [comment]
@Shiro-LK Thanks! Your repo is a great help to visualize using Tensorboard.

**Comment by baljit92** [comment]
@Shiro-LK Great repo. Helped me alot. 

However, when I try to train, I get an error 

> argument of type 'ResNetBackbone' is not iterable

Am i missing something?

**Comment by baljit92** [comment]
@Shiro-LK Great repo. Very helpful.

However, when i try to train by running the `train.py` file, all the losses `regression_loss`, `classification_loss` and `loss` are 0. I have attached a screenshot along.

This is the command i run : `python train.py --steps 100 --epochs 10 --weights ../../snapshots/resnet50_coco_best_v2.1.0.h5 --freeze-backbone csv ../../data/raccoon_dataset_train.csv ../../data/raccoon_dataset_classes.csv 
`
![screenshot from 2018-10-07 02-07-26](https://user-images.githubusercontent.com/5225501/46576580-c67b8180-c9d5-11e8-8975-f7ef6f937e65.png)
